### PR TITLE
MultiplexResolver now reports "not quantifiable" peptide features 

### PR DESCRIPTION
**Pull request by lars20070**
The combination of `FeatureFinderMultiplex + MultiplexResolver` allows the detection of true *knock-out* quantifications [1]. Take for example a simple light-heavy SILAC experiment. Say `FeatureFinderMultiplex` detects only a single peptide feature in the heavy channel. `MultiplexResolver` then completes the multiplet by adding a light *dummy peptide feature* of intensity `zero`. But this is not always correct.

There might be **two** distinct reasons, why a peptide feature singlet (and not a doublet) was detected.
(1) In the region of the expected light peptide feature is no peptide MS1 signal. In this case, a dummy feature of intensity `zero` is reported.
(2) Some isotopes of the light peptide feature overlap exactly with another unrelated peptide feature. `FeatureFinderMultiplex` blacklists this region and can consequently not detect the light peptide feature. In this case, a dummy feature of intensity `nan` is reported. The feature is *not quantifiable*. In complex, not fractionated samples, this case is increasingly likely [2].

This PR introduces the new (optional) `FeatureFinderMultiplex` output `-out_blacklist`. If this output is passed to the (optional) `MultiplexResolver` input `-in_blacklist`, then dummy features of intensity `nan` are reported. See (2) above.

[1] An absent peptide is reported as true `zero`, not as the sum of the *background noise* in this region. Say a gene/protein is knocked out in one of the two samples.

[2] A large number of `nan` are a sign that the sample is too complex. The sample should be further pre-fractionated or a longer LC gradient be used. Reliable peptide quantification is no longer possible.

### CODE REVIEW

File: src/topp/FeatureFinderMultiplex.cpp [hunk 1]
```diff
@@ -195,6 +199,18 @@ class TOPPFeatureFinderMultiplex :
     file.store(filename, map);
   }
   
+  /**
+   * @brief Write blacklist to mzML file.
+   *
+   * @param filename    name of mzML file
+   * @param blacklist    blaclist for output
+   */
+  void writeBlacklist_(const String& filename, MSExperiment& blacklist) const
```
Comment (hroest) [inline src/topp/FeatureFinderMultiplex.cpp]:
```suggestion
  void writeBlacklist_(const String& filename, const MSExperiment& blacklist) const
```

File: src/topp/FeatureFinderMultiplex.cpp [hunk 2]
```diff
@@ -195,6 +199,18 @@ class TOPPFeatureFinderMultiplex :
     file.store(filename, map);
   }
   
+  /**
+   * @brief Write blacklist to mzML file.
+   *
+   * @param filename    name of mzML file
+   * @param blacklist    blaclist for output
```
Comment (hroest) [inline src/topp/FeatureFinderMultiplex.cpp]:
```suggestion
   * @param blacklist    blacklist for output
```

**Comment by timosachsenberg** [comment]
@lars20070 can you clarify what happens to the unrelated feature that was successfully quantified before it led to blacklisting of the second feature?

**Comment by lars20070** [comment]
The unrelated feature is *lucky* and detected first in `FeatureFinderMultiplex`. It simply passes through `MultiplexResolver` unchanged.

**Comment by timosachsenberg** [comment]
the files added to this PR are too large for inclusion

**Comment by timosachsenberg** [comment]
@lars20070 out of interest. how often does this happen in complex samples?

**Comment by lars20070** [comment]
@timosachsenberg More than I had hoped/expected. I run tests on various samples and show you numbers at the retreat.

**Comment by lars20070** [comment]
@timosachsenberg The `mzML` was the huge file which is now reduced. Hope the other file sizes are ok.

File: src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderMultiplexAlgorithm.cpp [hunk 3]
```diff
@@ -1151,4 +1152,9 @@ namespace OpenMS
   {
     return(consensus_map_);
   }
+  
+  MSExperiment& FeatureFinderMultiplexAlgorithm::getBlacklist()
+  {
+    return(exp_blacklist_);
```
Comment (timosachsenberg) [inline src/openms/source/TRANSFORMATIONS/FEATUREFINDER/FeatureFinderMultiplexAlgorithm.cpp]: you might want to remove these extra braces in return statements:
return exp_blacklist_; is less redundant

**Comment by hroest** [comment]
> @timosachsenberg The `mzML` was the huge file which is now reduced. Hope the other file sizes are ok.

@timosachsenberg to make sure this doesnt appear in the git history, you will need to squash the commits (instead of merging)

**Comment by timosachsenberg** [comment]
rebuild jenkins

**Comment by timosachsenberg** [comment]
Ok I will merge. But imho you should add some extra documentation that this is mainly kockout related

**Comment by lars20070** [comment]
@timosachsenberg In this example dataset, 5% of all reported features are not quantifiable, see [table 3](https://github.com/ProteomicsScripts/ProteomicsScripts/blob/master/mzTab2report/misc/example_7.pdf). This explain (nearly) all outliers in the peptide fold changes for specific proteins, see before/after plots below.

[plot_ProteinsManyPeptides_1_BEFORE.pdf](https://github.com/OpenMS/OpenMS/files/2961604/plot_ProteinsManyPeptides_1_BEFORE.pdf)
[plot_ProteinsManyPeptides_1_AFTER.pdf](https://github.com/OpenMS/OpenMS/files/2961605/plot_ProteinsManyPeptides_1_AFTER.pdf)

**Pull request merged**